### PR TITLE
[COMPASS-1785] Support Electron Renderer Tests

### PR DESCRIPTION
**Pull request by matt-d-rat**
This PR adds support for running tests in an electron renderer context inside electron by utilising Karma.

Previously this support was provided by hadron-build, via [electron-mocha](https://github.com/jprichardson/electron-mocha) - but given we now have a more complicated pre-compilation step requiring webpack, electron-mocha doesn't facilitate a means for us to achieve this. Hence the switch to Karma (facilitated by [karma-webpack](https://github.com/webpack-contrib/karma-webpack) and [karma-electron](https://github.com/twolfson/karma-electron)).

Renderer tests (for the time being) live under `test/renderer` as there is otherwise an issue with chai-enzyme usage within our current unit tests that are run with mocha-webpack. A subsequent PR will be raised to solve this issue, and allow us to run our entire test suite through the electron-renderer as well.

Down the road, testing should hopefully be centralised back to hadron-build once all existing plugins have been upgraded to the new template. This is a stop gap on the road to achieving that goal.

As part of this PR I also completed the following chores:

- Converted Actions and Stores over to ES6.
- `index.js` in `stores` and `actions` now is simply an import/export file, the implementation of an action/store has been moved to its own file to align with new documentation on best practices.
- Added example renderer test to `test/renderer`.
- Updated package.json with new dependencies.
- Added new test command `npm run test:karma` to package.json.
- Configured Karma.
- Added new common aliases to webpack configuration files.

### CODE REVIEW

File: template/test/renderer/Store.spec.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import Store from 'stores';
+
+describe('{{pascalcase name}}Store [Store]', () => {
+  beforeEach(function() {
```
Comment (rueckstiess) [inline template/test/renderer/Store.spec.js]: these tests are just a copy of the Store.spec.js file, right? Might be confusing to users of this plugin. Can we think of a test that actually needs to be executed in the renderer process instead?

File: template/karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const webpackConfig = require('./config/webpack.karma.config');
+
+module.exports = function(config) {
+	config.set({
```
[the remark this answers is not in the document]
Comment (matt-d-rat) [reply template/karma.conf.js]: - [ ] Fix code style

File: template/src/actions/Actions.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+const {{pascalcase name}}Actions = Reflux.createActions([
+  /**
+   * define your actions as strings below, for example:
+   */
+  'toggleStatus'
+]);
+
+export default {{pascalcase name}}Actions;
+export { {{pascalcase name}}Actions };
```
[the remark this answers is not in the document]
Comment (matt-d-rat) [reply template/src/actions/Actions.js]: - [ ] Fix code style

File: template/src/stores/index.js [hunk 4]
```diff
@@ -1,91 +1,4 @@
...
-   * @param  {Object} prevState   previous state.
-   */
-  storeDidUpdate(prevState) {
-    debug('{{pascalcase name}} store changed from', prevState, 'to', this.state);
-  }
-});
-
-module.exports = {{pascalcase name}}Store;
+export default {{pascalcase name}}Store;
+export { {{pascalcase name}}Store };
```
[the remark this answers is not in the document]
Comment (matt-d-rat) [reply template/src/stores/index.js]: - [ ] Fix code style

File: template/src/stores/Store.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
+import Reflux from 'reflux';
```
[the remark this answers is not in the document]
Comment (matt-d-rat) [reply template/src/stores/Store.js]: In the case of React it's an Eslint AirBnb code style guide requirement to use PascalCase for imports in the case of classes (http://airbnb.io/javascript/#naming--PascalCase). I assumed, and possibly incorrectly that the default import of Reflux was also a class, or at least an instantiated object.

TL;DR, I don't have a good reason for or against. Based on our code style guides what you prefer the syntax to resemble.

File: template/test/renderer/Store.spec.js (hunk 1 above)
Comment (matt-d-rat) [reply template/test/renderer/Store.spec.js]: @rueckstiess I agree, and given more time I could probably come up with a more contrived example to better demonstrate, but the important thing was simply having something there to test to demonstrate the Karma-Electron renderer tests run and pass.

If you're ok with leaving this as is for now, I can address this comment with a better example as part of another PR when I have more time and fewer higher priority items to get through. How does that sound?

File: template/src/stores/Store.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (rueckstiess) [reply template/src/stores/Store.js]: Discussed amongst Sydney devs and majority would prefer to keep all file names [slug-case](https://github.com/ianstormtaylor/to-slug-case) (i.e. all lower-case and with dashes).

Reasons for the change:
- reduced mental mapping from slug-case to UpperCamelCase (the component name)
- webpack can warn of case-insensitive imports

Reasons against change:
- all other mongodb-js modules follow this convention
- potential [issues](https://github.com/airbnb/javascript/issues/501) with case-insensitive file systems

File: template/test/renderer/Store.spec.js (hunk 1 above)
Comment (rueckstiess) [reply template/test/renderer/Store.spec.js]: ok, let's leave as is. But could you add a comment on the top of the file, something like this (feel free to change the wording).

```
/*
 * Place tests that must run in a renderer context inside Electron here. 
 * 
 * Note: The tests below are just a copy of the store unit tests as an example. 
 * More complex plugins will require actual renderer/integration tests to be
 * executed here.
 */

Comment (matt-d-rat) [reply template/test/renderer/Store.spec.js]: @rueckstiess Yeah that's a good idea, I'll add the comment and then return to this in another PR.

File: template/src/stores/Store.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (matt-d-rat) [reply template/src/stores/Store.js]: - [ ] Update the filenames (use `git mv` to achieve this otherwise git won't recognise the change in name in the case of case insensitive filesystems like Mac OSX)
